File: UserClienJava/app/src/main/java/com/pacarius/userclienjava/MapTest.java
package com.pacarius.userclienjava;

import android.os.Bundle;
import android.text.InputType;
import android.widget.EditText;
import android.widget.ImageView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MapTest extends AppCompatActivity {
    ImageView iv;
    MapTestPlot mv;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_map_test);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        iv = findViewById(R.id.mapImage);
        iv.setImageDrawable(getDrawable(R.drawable.street));
        iv.requestLayout();
        mv = findViewById(R.id.mapTestPlotView);
        mv.setImageView(iv);
        setIp();
        mv.redraw();
    }

    private void setIp() {
        EditText input = new EditText(this);
        input.setInputType(InputType.TYPE_CLASS_TEXT);

        AlertDialog dialog = new AlertDialog.Builder(this)
                .setTitle("Change IP")
                .setMessage("Enter the new IP address:")
                .setView(input)
                .setPositiveButton("OK", (dialog1, which) -> {
                    String newIp = input.getText().toString();
                    mv.handler().setIp(newIp);
                })
                .setNegativeButton("Cancel", null)
                .create();

        dialog.show();
    }
}

File: UserClient/app/src/main/java/com/pacarius/client/MapTest.kt
package com.pacarius.client

import android.os.Bundle
import android.text.InputType
import android.view.ViewTreeObserver
import android.widget.EditText
import android.widget.ImageView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MapTest : AppCompatActivity() {
    lateinit var iv: ImageView
    lateinit var mv: MapTestPlot
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_map_test)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        iv = findViewById(R.id.mapImage)
        iv.setImageDrawable(getDrawable(R.drawable.street))
        iv.requestLayout()
        mv = findViewById(R.id.mapTestPlotView)
        mv.imageView = iv
        setIp()
        mv.redraw()
    }
    private fun setIp(){
        val input = EditText(this)
        input.inputType = InputType.TYPE_CLASS_TEXT

        // Create an AlertDialog
        val dialog = AlertDialog.Builder(this)
            .setTitle("Change IP")
            .setMessage("Enter the new IP address:")
            .setView(input)
            .setPositiveButton("OK") { _, _ ->
                // Get the text entered by the user
                val newIp = input.text.toString()

                // Pass the new IP address to the MapTestPlot view
                mv.handler.ip = newIp
            }
            .setNegativeButton("Cancel", null)
            .create()

        // Show the dialog
        dialog.show()
    }
}